specs/Extensions.md

```diff
@@ -0,0 +1,579 @@
+# Background
+Developers would like to ensure a browser extension is installed so that they may take advantage of the functionality the extension is providing.
+
+# Description
+Enable extension services in WebView2. Then end developer can `AddBrowserExtension` from folder path, `GetBrowserExtension` to get list of extensions installed. Once an extension is installed,
+developers can use `ICoreWebView2BrowserExtension` to get id, name of this extension. And also remove or enable/disable this extension.
+
+# Examples
+
+The following code snippet demonstrates how the Extensions related API can be used:
+
+## Install extension on first run
+
+You may have a browser extension that is specific to your application's web content. This snippet shows how you can install your browser extension on your first run.
+
+### Win32 C++
+```cpp
+static constexpr WCHAR m_defaultExtensionFolderPath = L"//Mydrive";
+
+void AppWindow::InitializeWebView()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    CHECK_FAILURE(options->put_AreBrowserExtensionsEnabled(TRUE));
+
+    // ... other option properties
+
+    // ... CreateCoreWebView2EnvironmentWithOptions
+
+    InstallDefaultExtensions();
+    
+    // ... Navigate
+}
+
+void ScenarioExtensionsManagement::InstallDefaultExtensions()
+{
+    auto webView2_13 = m_webView.try_query<ICoreWebView2_13>();
+    CHECK_FEATURE_RETURN_EMPTY(webView2_13);
+    wil::com_ptr<ICoreWebView2Profile> webView2Profile;
+    CHECK_FAILURE(webView2_13->get_Profile(&webView2Profile));
+    auto profile6 = webView2Profile.try_query<ICoreWebView2Profile6>();
+    CHECK_FEATURE_RETURN_EMPTY(profile6);
+
+    profile6->GetBrowserExtensions(
+        Callback<ICoreWebView2ProfileGetBrowserExtensionCompletedHandler>(
+            [this, profile6, extension_path](
+                HRESULT error, ICoreWebView2BrowserExtensionList* extensions) -> HRESULT
+            {
+                std::wstring extensionIdString;
+                bool extensionInstalled = false;
+                UINT extensionsCount = 0;
+                extensions->get_Count(&extensionsCount);
+
+                for (UINT index = 0; index < extensionsCount; ++index)
+                {
+                    wil::com_ptr<ICoreWebView2BrowserExtension> extension;
+                    extensions->GetValueAtIndex(index, &extension);
+
+                    wil::unique_cotaskmem_string id;
+                    std::wstring message;
+
+                    extension->get_Id(&id);
+                    extensionIdString = id.get();
+
+                    if (extensionIdString.compare(m_extensionId) == 0)
+                    {
+                        extensionInstalled = true;
+                        break;
+                    }
+                }
+
+                if (!extensionInstalled)
+                {
+                    CHECK_FAILURE(profile6->AddBrowserExtension(
+                        m_defaultExtensionFolderPath.c_str(),
```

Comment:
Above m_defaultExtensionFolderPath is a WCHAR but c_str() would suggest its like a std::string or something?